core/src/main/java/org/bitcoinj/core/VersionedChecksummedBytes.java

```diff
@@ -33,29 +36,20 @@
  * <p>and the result is then Base58 encoded. This format is used for addresses, and private keys exported using the
  * dumpprivkey command.</p>
  */
-public class VersionedChecksummedBytes implements Serializable, Cloneable, Comparable<VersionedChecksummedBytes> {
-    protected final int version;
-    protected byte[] bytes;
-
-    protected VersionedChecksummedBytes(String encoded) throws AddressFormatException {
-        byte[] versionAndDataBytes = Base58.decodeChecked(encoded);
-        byte versionByte = versionAndDataBytes[0];
-        version = versionByte & 0xFF;
-        bytes = new byte[versionAndDataBytes.length - 1];
-        System.arraycopy(versionAndDataBytes, 1, bytes, 0, versionAndDataBytes.length - 1);
-    }
+public abstract class VersionedChecksummedBytes implements Serializable, Cloneable, Comparable<VersionedChecksummedBytes> {
+    protected final transient NetworkParameters params;
+    protected final byte[] bytes;
 
-    protected VersionedChecksummedBytes(int version, byte[] bytes) {
-        checkArgument(version >= 0 && version < 256);
-        this.version = version;
-        this.bytes = bytes;
+    protected VersionedChecksummedBytes(NetworkParameters params, byte[] bytes) {
+        this.params = checkNotNull(params);
+        this.bytes = checkNotNull(bytes);
     }
 
     /**
      * Returns the base-58 encoded String representation of this
      * object, including version and checksum bytes.
      */
-    public final String toBase58() {
+    protected static String toBase58(int version, byte[] bytes) {
```

Comment:
Maybe this could go to `Base58` as `encodeChecked`?

Reply:
Good idea, I implemented this via b87d22e2beb328b140be84b5b7e3eced4637b4b3 on master.